### PR TITLE
Refactor jupyter uri picker13893

### PR DESCRIPTION
**Pull request by DonJayamanne**
For #13892

### CODE REVIEW

File: src/standalone/userJupyterServer/jupyterPasswordConnect.ts [hunk 1]
```diff
@@ -52,6 +58,8 @@ export class JupyterPasswordConnect {
         isTokenEmpty: boolean;
         displayName?: string;
         handle: string;
+        validationErrorMessage?: string;
```
Comment (DonJayamanne) [inline src/standalone/userJupyterServer/jupyterPasswordConnect.ts:61]: Temporary hacks until we support both old an new APIs.

File: src/standalone/userJupyterServer/jupyterPasswordConnect.ts [hunk 2]
```diff
@@ -282,14 +302,28 @@ export class JupyterPasswordConnect {
...
+    /**
+     * The input prompts created here are not disposed and hidden immediately.
+     * The idea is that the workflow that requires this method to prompt for password
+     * will return with the prompt displayed and then if the password is invalid or the like,
+     * we can call this method once again and display a new quick pick avoiding flickers.
+     *
+     * Similarly, if there's another quick pick or input box that needs to be displayed after this method,
+     * leaving this UI open will avoid flickers.
+     *
+     * The disposables array is eventually disposed by the calling method.
```
Comment (DonJayamanne) [inline src/standalone/userJupyterServer/jupyterPasswordConnect.ts:314]: Temporary hacks until we support both old an new APIs.

File: src/kernels/jupyter/connection/serverSelector.ts [hunk 3]
```diff
@@ -27,47 +19,6 @@ export type SelectJupyterUriCommandSource =
     | 'errorHandler'
     | 'prompt';
 
-export async function validateSelectJupyterURI(
```
Comment (DonJayamanne) [inline src/kernels/jupyter/connection/serverSelector.ts:30]: Dead code

File: src/standalone/userJupyterServer/userServerUrlProvider.ts [hunk 4]
```diff
@@ -77,39 +73,37 @@ const GlobalStateUserAllowsInsecureConnections = 'DataScienceAllowInsecureConnec
...
 @injectable()
 export class UserJupyterServerUrlProvider
-    implements
-        IExtensionSyncActivationService,
-        IDisposable,
-        IInternalJupyterUriProvider,
-        JupyterServerProvider,
-        JupyterServerCommandProvider
+    extends Disposables
+    implements IExtensionSyncActivationService, IDisposable, JupyterServerProvider, JupyterServerCommandProvider
```
Comment (DonJayamanne) [inline src/standalone/userJupyterServer/userServerUrlProvider.ts:77]: Temporary hacks to support multipllle interfaces as we support both old an new APIs, once we remove the old API, this class will get much smaller

File: src/standalone/userJupyterServer/userServerUrlProvider.ts [hunk 5]
```diff
@@ -354,314 +347,430 @@ export class UserJupyterServerUrlProvider
...
-                        } else {
-                            traceError(`Failed to get the password for ${jupyterServerUri.baseUrl}`, err);
-                            if (inputWasHidden) {
-                                input.show();
+                    // Capture the password disposable objects into a separate array.
+                    // We want to keep this Quick Pick UI open, as the validation happens outside that class.
+                    // The UI gets closed when we dispose the disposables.
+                    // After the validation, we dispose the UI, and optionally display a new UI.
+                    // This way there is no flicker in the UI.
+                    const passwordDisposables: Disposable[] = [];
```
Comment (DonJayamanne) [inline src/standalone/userJupyterServer/userServerUrlProvider.ts:409]: This variable is a temporary hack, until we remove the old API

**Review by connor4312 — APPROVED** [review]

**Pull request merged**